bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -55,20 +54,19 @@
 
     static final Logger LOGGER = Loggers.getLogger("ObjectId");
 
+    private static final int OBJECT_ID_LENGTH = 12;
     private static final int LOW_ORDER_THREE_BYTES = 0x00ffffff;
+    private static final int RANDOM_VALUE_LENGTH = 5;
 
-    private static final int MACHINE_IDENTIFIER;
-    private static final short PROCESS_IDENTIFIER;
     private static final AtomicInteger NEXT_COUNTER = new AtomicInteger(new SecureRandom().nextInt());
 
     private static final char[] HEX_CHARS = new char[] {
-      '0', '1', '2', '3', '4', '5', '6', '7',
-      '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
+            '0', '1', '2', '3', '4', '5', '6', '7',
+            '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
 
     private final int timestamp;
-    private final int machineIdentifier;
-    private final short processIdentifier;
     private final int counter;
+    private final byte[] randomValue = new byte[RANDOM_VALUE_LENGTH];
```

Comment:
I'd like to keep away from an extra allocation here.  Can we stick two primitives and just change the way they are initialized?

Reply:
I now use two ints to represent the 5-byte random value.